octopusdeploy3-single-vm-windows/azuredeploy.json

```diff
@@ -0,0 +1,355 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+      "defaultValue": "West US",
+      "allowedValues": [
+        "Central US",
+        "East Asia",
+        "East US",
+        "East US 2",
+        "Japan East",
+        "Japan West",
+        "North Europe",
+        "South Central US",
+        "Southeast Asia",
+        "West Europe",
+        "West US"
+      ],
+      "metadata": {
+        "Description": "Location of the Octopus Deploy server."
+      }
+    },
+    "vmStorageAccountName": {
+      "type": "string",      
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account that will hold the Octopus Deploy Virtual Machine Disks."
+      }
+    },
+    "vmAdminUsername": {
+      "type": "string",
+      "defaultValue": "octoadmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy Virtual Machine."
+      }
+    },
+    "networkDnsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name used to access the Octopus Deploy server via HTTP or RDP."
+      }
+    },
+    "sqlServerName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the SQL DB Server that will hold the Octopus Deploy data."
+      }
+    },
+    "sqlAdminUsername": {
+      "type": "string",
+      "defaultValue": "sqladmin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "sqlAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy SQL DB Server."
+      }
+    },
+    "licenseFullName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Full Name."
+      }
+    },
+    "licenseOrganisationName": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Organisation Name."
+      }
+    },
+    "licenseEmailAddress": { 
+      "type": "string",
+      "metadata": {
+        "description": "Octopus Deploy Trial license - provide Email Addres."
+      }
+    },
+    "octopusAdminUsername" : {
+      "type": "string",
+      "defaultValue": "admin",
+      "metadata": {
+        "description": "Admin username for the Octopus Deploy web application."
+      }
+    },
+    "octopusAdminPassword" : {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password for the Octopus Deploy web application."
+      }
+    }
+  },
+  "variables": {
+    "location": "[parameters('location')]",
```

Comment:
This is an approach to abstracting the method of obtaining a value for later use in the template. This allows you to simply use variables in the template for reference. You can decide in the variable section whether you define the value there or obtain it from a parameter. 

This definitely helped me in the stages when I was defining the template and switching between obtaining a value from a parameter or a variable. It may still have value beyond that and it may still be worth showing this as an alternate method for defining parameters and variables.